### PR TITLE
mvcc: remove unused restore method

### PR DESCRIPTION
**Pull request by sinsharat**
Removing unused method (*treeIndex) Restore() method as asked by @xiang90 in PR #6846

### CODE REVIEW

File: mvcc/index_test.go [hunk 1]
```diff
@@ -252,13 +256,15 @@ func TestIndexCompact(t *testing.T) {
 		}
 		am := ti.Compact(i)
 
-		wti := newTreeIndex()
+		wti := &treeIndex{
```
Comment (xiang90) [inline mvcc/index_test.go]: same here?

**Comment by sinsharat** [comment]
@xiang90 i have done the changes. PTAL.
Thank you!

**Comment by xiang90** [comment]
lgtm. defer to @heyitsanthony

**Comment by heyitsanthony** [comment]
lgtm

**Pull request merged**